net.go

```diff
@@ -425,6 +426,122 @@ func (e *EurekaConnection) ScheduleSecureVIPAddressUpdates(addr string, await bo
 	return e.scheduleVIPAddressUpdates(addr, await, done, options), nil
 }
 
+// An InstanceSetSource holds a periodically updated set of instances registered with Eureka.
+type InstanceSetSource struct {
+	m         *sync.RWMutex
+	instances []*Instance
+	done      chan<- struct{}
+}
+
+func newInstanceSetSourceFromChannel(await bool, updates <-chan InstanceSetUpdate, done chan<- struct{}) *InstanceSetSource {
```

Comment:
Please leave it as `boolean`. I would like to reduce number of exposed types. Thank you for your consideration and research.